nsxt/resource_nsxt_logical_tier0_router.go

```diff
@@ -0,0 +1,176 @@
+/* Copyright © 2017 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	api "github.com/vmware/go-vmware-nsxt"
+	"github.com/vmware/go-vmware-nsxt/manager"
+	"log"
+	"net/http"
+)
+
+var highAvailabilityValues = []string{"ACTIVE_ACTIVE", "ACTIVE_STANDBY"}
+
+// TODO: add advanced config
+func resourceNsxtLogicalTier0Router() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtLogicalTier0RouterCreate,
+		Read:   resourceNsxtLogicalTier0RouterRead,
+		Update: resourceNsxtLogicalTier0RouterUpdate,
+		Delete: resourceNsxtLogicalTier0RouterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"revision": getRevisionSchema(),
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "Description of this resource",
+				Optional:    true,
+			},
+			"display_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The display name of this resource. Defaults to ID if not set",
+				Optional:    true,
+				Computed:    true,
+			},
+			"tag": getTagsSchema(),
+			"high_availability_mode": &schema.Schema{
+				Type:         schema.TypeString,
+				Description:  "High availability mode",
+				Default:      "ACTIVE_STANDBY",
+				Optional:     true,
+				ForceNew:     true, // Cannot change the HA mode of a router
+				ValidateFunc: validation.StringInSlice(highAvailabilityValues, false),
+			},
+			"firewall_sections": getResourceReferencesSchema(false, true, []string{}, "List of Firewall sections related to the logical router"),
+			"edge_cluster_id": &schema.Schema{
```

Comment:
Maybe this one is effectively ForceNew (I think you can't change edge cluster if its connected up)

Reply:
Done